### ISSUE TITLE
feat(linter): eslint/no-import-assign (partially done, in nursery category)

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#610]
Implement ESLint's [`no-import-assign`](https://eslint.org/docs/latest/rules/no-import-assign) rule.

To get started, run
```sh
just new-rule no-import-assign
```

**Comment by DeividAlmeida** [comment]
hi, i want to continue this rule, but i saw that it's implemented, what can i do to conclude ?

**Comment by mysteryven** [comment]
> hi, i want to continue this rule, but i saw that it's implemented, what can i do to conclude ?

There are [some cases](https://github.com/oxc-project/oxc/blob/c6eb519417d4d6328a6a78de861f706541f56a39/crates/oxc_linter/src/rules/eslint/no_import_assign.rs#L158-L182) that are not supported yet, we can try to pass more!

**Comment by DeividAlmeida** [comment]
should i open a else block to deal with a Read end ReadWrite flag in no_import_assign.rs?

**Comment by DeividAlmeida** [comment]
what approach do you suggest in this case?

**Comment by Boshen** [comment]
> should i open a else block to deal with a Read end ReadWrite flag in no_import_assign.rs?

You may give it a try 😁

**Comment by DeividAlmeida** [comment]
To improve this should improve all its references like @Boshen said here [#2164 ](https://github.com/oxc-project/oxc/pull/2164)

**Comment by Boshen** [comment]
I think we can port this exactly https://github.com/eslint/eslint/blob/972ef155a94ad2cc85db7d209ad869869222c14c/lib/rules/no-import-assign.js#L204-L206
